irohad/ametsuchi/key_value_storage.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_KV_STORAGE_HPP
+#define IROHA_KV_STORAGE_HPP
+
+#include <boost/optional.hpp>
+#include <string>
+#include <vector>
+
+namespace iroha {
+
+  namespace ametsuchi {
+
+    /**
+     * Solid storage interface
+     */
+    class KeyValueStorage {
+     public:
+      /**
+       * Type of storage key
+       */
+      using Identifier = uint32_t;
+
+      /**
+       * Add entity with binary data
+       * @param id - reference key
+       * @param blob - data associated with key
+       */
+      virtual bool add(Identifier id, const std::vector<uint8_t> &blob) = 0;
+
+      /**
+       * Get data associated with
+       * @param id - reference key
+       * @return - blob, if exists
+       */
+      virtual boost::optional<std::vector<uint8_t>> get(
+          Identifier id) const = 0;
+
+      /**
+       * @return folder of storage
+       */
+      virtual std::string directory() const = 0;
+
+      /**
+       * @return maximal not null key
```

Comment:
Comment is not well worded. It implies a total ordering of the keys, but the name of the function last_id implies chronological ordering. Thus interface contradicts documentation.

For example if we add keys in this order: 1 -> 3 -> 2. Name of the function suggests that key 2 is returned, but comment tells us that we should expect 3.

Reply:
@return chronologically last not null key?